swig/python/gdal-utils/osgeo_utils/gdal2tiles.py

```diff
@@ -48,7 +48,11 @@
 import tempfile
 import threading
 from functools import partial
-from multiprocessing import Pool
+if 'mpi4py.futures' in sys.modules:
```

Comment:
I'm completely ignorant about mpi4py.futures. Just wondering if there would be use cases where mpi4py.futures would be in sys.module and where using that feature wouldn't be needed, e.g someone using gdal2tiles not as a script but as a function, in a Python session where mpi4py.futures would be loaded ? That's maybe unlikely ?

Otherwise it would be nice to document that new capability in doc/source/programs/gdal2tiles.rst

Reply:
It seems very unlikely that someone calling osgeo_utils.gdal2tiles.main() directly, which is undocumented, would also be using the mpi4py.futures module (which is only likely on an HPC cluster) but not want gdal2tiles to also use the mpi4py.futures module. That said, I can refactor the module to check for an "--mpi" flag and then also modify the script to load mpi4py.futures internally so that rather than "mpiexec ... python -m mpi4py.futures -m osgeo_utils.gdal2tiles ..." the launch command becomes "mpiexec ... gdal2tiles.py --mpi ...".

Reply:
> That said, I can refactor the module to check for an "--mpi" flag and then also modify the script to load mpi4py.futures internally so that rather than "mpiexec ... python -m mpi4py.futures -m osgeo_utils.gdal2tiles ..." the launch command becomes "mpiexec ... gdal2tiles.py --mpi ...".

that sounds good